CONTRIBUTING.rst

```diff
@@ -19,8 +22,59 @@ Contributor Guide
 
    * Now, you have remote repositories named:
 
-      - ``upstream``, which refers to the ``networkx`` repository
-      - ``origin``, which refers to your personal fork
+     - ``upstream``, which refers to the ``networkx`` repository
+     - ``origin``, which refers to your personal fork
+
+   * Next, you need to set up your build environment.
+     Here are instructions for two popular environment managers:
+   
+     * ``venv`` (pip based)
+     
+       ::
+     
+         # Create a virtualenv named ``networkx-dev`` that lives in the directory of
+         # the same name
+         python -m venv networkx-dev
+         # Activate it
+         source networkx-dev/bin/activate
+         # Install main development and runtime dependencies of networkx
+         pip install -r <(cat requirements/{default,developer,doc,optional,test}.txt)
+         #
+         # (Optional) Install pygraphviz, pydot, and gdal packages
+         # These packages require that you have your system properly configured
+         # and what that involves differs on various systems.
+         # pip install -r requirements/extras.txt
+         #
+         # Build and install networkx from source
+         pip install -e .
+         # Test your installation
+         PYTHONPATH=. pytest networkx
+     
+     * ``conda`` (Anaconda or Miniconda)
+    
+       ::
+ 
+         # Create a conda environment named ``networkx-dev``
+         conda create --name networkx-dev
+         # Activate it
+         conda activate networkx-dev
+         # Install main development and runtime dependencies of networkx
+         conda install `for i in requirements/{default,developer,doc,optional,test}.txt; do echo -n " --file $i "; done`
```

Comment:
We should probably suggest
```python
conda install -c conda-forge `for i in requirements/{default,developer,doc,optional,test}.txt; do echo -n " --file $i "; done`
```
at least until the default channel expands beyond providing only graphviz v2.40.

Switching to suggesting brew can also work. I chose conda over brew a few years ago because conda seemed more connected to scientific packages and it seemed much lighter weight in terms of extra packages it installed and time it took to install. brew may be better now.